ci/run-tests.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+set -e
+
+VERSION="${TRAVIS_TAG:-0.0.0}"
+VERSION="${VERSION#v}"
+
+readonly ROOT_DIR="$(dirname "$(dirname "${0}")")"
+cd "${ROOT_DIR}"
+
+function test_fluentd_plugin() {
+  local plugin_name="${1}"
+  local version="${2}"
+  # Strip everything after "-" (longest match) to avoid gem prerelease behavior
+  local gem_version="${version%%-*}"
+
+  cd "${plugin_name}" || exit 1
+
+  if [[ "${version}" == "" ]] ; then
+    echo "Please provide the version when bundling fluentd plugins"
+    exit 1
+  fi
+
+  echo "Preparing gem ${plugin_name} version ${gem_version} in $(pwd) for testing ..."
+  sed -i.bak "s/0.0.0/${gem_version}/g" ./"${plugin_name}".gemspec
+  rm -f ./"${plugin_name}".gemspec.bak
+
+  echo "Install bundler..."
+  bundle install
+
+  echo "Run unit tests..."
+  bundle exec rake || exit 1
+}
+
+function test_fluentd_plugins() {
+  local version="${1}"
+
+  if [[ "${version}" == "" ]] ; then
+    echo "Please provide the version when bundling fluentd plugins"
+    exit 1
+  fi
+
+  find . -name 'fluent-plugin-*' -type 'd' -print |
+    while read -r line; do
+      # Run tests in their own context
+      (test_fluentd_plugin "$(basename "${line}")" "${version}") || exit 1
+    done
+}
+
+## check the build scripts with shellcheck
+echo "Checking the build script with shellcheck..."
+shellcheck ci/build.sh
+shellcheck ci/run-tests.sh
```

Comment:
Add option to enable all checks, my suggestion:

```suggestion
# Enable all checks but exclude SC2154 which complains about undeclared but used variables
# that we get from our CI environment e.g. ${TRAVIS_EVENT_TYPE} and ${TRAVIS_PULL_REQUEST_BRANCH}
shellcheck --enable all --exclude SC2154 ci/build.sh
shellcheck --enable all --exclude SC2154 ci/run-tests.sh
```

Reply:
Rather than excluding we should just take care of that and set those variables to some default, eg.
```TRAVIS_EVENT_TYPE="${TRAVIS_EVENT_TYPE:-push}"```

Reply:
I disagree. `TRAVIS_EVENT_TYPE` doesn't make sense when not on travis and this script is not intended **ONLY** to be run on travis.

Reply:
If it's not intended then those variables should be extracted somewhere else.
Anyway disabling globally shellcheck rule for something that is easily fixed is not a good idea IMO.

Reply:
Let's disable it in lines before the "known issues"

Reply:
```
Checking the build script with shellcheck...
unrecognized option `--enable'

Usage: shellcheck [OPTIONS...] FILES...
  -e CODE1,CODE2..  --exclude=CODE1,CODE2..  exclude types of warnings
  -f FORMAT         --format=FORMAT          output format
  -C[WHEN]          --color[=WHEN]           Use color (auto, always, never)
  -s SHELLNAME      --shell=SHELLNAME        Specify dialect (sh,bash,dash,ksh)
  -x                --external-sources       Allow 'source' outside of FILES.
  -V                --version                Print version information
```

Reply:
:/ Let's leave it then without `--enable`

Reply:
I will update shellcheck in vagrant